authzed/api/v1/core.proto

```diff
@@ -21,6 +22,24 @@ message Relationship {
 
   // subject is the subject to which the resource is related, in some manner.
   SubjectReference subject = 3 [ (validate.rules).message.required = true ];
+
+  // caveat is a reference to a the caveat that must be enforced over the relationship
+  ContextualizedCaveat optional_caveat = 4 [ (validate.rules).message.required = false ];
+}
+
+/**
+ * ContextualizedCaveat represents a reference to a caveat used to by caveated tuples.
+ * The context are key-value pairs that will be injected at evaluation time.
+ */
+message ContextualizedCaveat {
+  /** caveat_name is the name used in the schema for a stored caveat **/
```

Comment:
`caveat_name` is the name of the caveat expression to use, as defined in the schema